replica/database.cc

```diff
@@ -2221,6 +2221,18 @@ schema_ptr database::find_indexed_table(const sstring& ks_name, const sstring& i
 
 future<> database::close_tables(table_kind kind_to_close) {
     auto b = defer([this] { _stop_barrier.abort(); });
+
+    if (kind_to_close == table_kind::system) {
+        // Stop compaction for all system tables before closing them in
+        // parallel to prevent any possible race between compaction stoppage
+        // and closure of compaction_history system table.
+        co_await _tables_metadata.parallel_for_each_table(coroutine::lambda([] (table_id, lw_shared_ptr<table> table) {
```

Comment:
why use `coroutine::lambda()` here? in the body of the lambda, instead of using C++20 coroutine, it is using the plain future-promise.

Reply:
Ack. I have fixed it now.